app/src/main/java/datastore/group_api/route/GroupRoutes.java

```diff
@@ -0,0 +1,89 @@
+package datastore.group_api.route;
+import java.util.List;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response.Status;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import datastore.group_api.database.GroupRepository;
+import datastore.group_api.entity.Group;
+
+@Path("/groups")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class GroupRoutes {
+    @Inject GroupRepository groupRepo;
+    @GET
+    public List<Group> getAll() {
+        return groupRepo.listAll();
+    }
+
+    @GET
+    @Path("{id}")
+    public Group getById(@PathParam("id") Long id) {
+        return groupRepo.findByIdOptional(id).orElseThrow(NotFoundException::new);
+    }
+    
+    @GET
+    @Path("name/{name}")
```

Comment:
I gave the same suggestion to the other group - I would suggest this be an optional query param on the `getAll` method. This is more a search criteria on the method to retrieve all groups.